transport/internal/tlsmux/listener.go

```diff
@@ -180,7 +188,15 @@ func (l *listener) handlePlaintextConn(conn net.Conn) net.Conn {
 }
 
 func matchTLSConnection(cs *connSniffer) (bool, error) {
-	// TODO(jronak): set temporary connection read and write timeout.
+	if err := cs.SetReadDeadline(time.Now().Add(_sniffReadTimeout)); err != nil {
+		return false, err
+	}
+
+	// Reset read deadline after sniffing.
+	defer func() {
+		_ = cs.SetReadDeadline(time.Time{})
```

Comment:
just `cs.SetReadDeadline(time.Time{})`?

Reply:
also why `time.Time{}` ?

Reply:
`time.Time{}` is more like a 0 value deadline basically meaning no timeout. 
See HTTP server for example: https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/net/http/server.go;l=1862

Reply:
let's link that in the comment :) looks good to me.